client/my-sites/stats/activity-log/utils.js

```diff
@@ -0,0 +1,26 @@
+/** @format */
+/**
+ * @typedef OffsetParams
+ * @property {?string} timezone  Timezone representation to apply.
+ * @property {?string} gmtOffset Offset to apply if timezone isn't supplied.
+ * @property {object}  moment    Moment object to which timezone or offset will be applied.
+ */
+
+/**
+ * Accepts nullable timezone and offset and applies one to the provided moment, preferring the
+ * timezone. If neither are provided, return the moment unchanged.
+ *
+ * @param  {OffsetParams} params Parameters
+ * @return {Object}       Moment with timezone applied if provided.
+ *                        Moment with gmtOffset applied if no timezone is provided.
+ *                        If neither is provided, the original moment is returned.
+ */
+export function adjustMoment( { timezone, gmtOffset, moment } ) {
```

Comment:
I merged before I saw this comment, but you're right. That would be a good place to start #17320